### PR TITLE
Metrics for Dynamo capacity

### PR DESCRIPTION
**Pull request by jml**
Part of https://github.com/weaveworks/monitoring/issues/12

---
This change is [Reviewable](https://reviewable.io/reviews/weaveworks/cortex/89)

### CODE REVIEW

File: cmd/cortex/main.go [hunk 1]
```diff
@@ -115,6 +115,12 @@ func main() {
 	if err != nil {
 		log.Fatalf("Error initializing chunk store: %v", err)
 	}
+	resourceWatcher, err := chunk.WatchDynamo(cfg.dynamodbURL, 2*time.Minute)
```
Comment (tomwilkie) [inline cmd/cortex/main.go]: `2*time.Minute` command line flag for this?

File: chunk/dynamo_client.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+func (w *dynamoWatcher) Describe(ch chan<- *prometheus.Desc) {
+	w.accountMaxCapacity.Describe(ch)
+	w.tableCapacity.Describe(ch)
+}
+
+// Collect implements prometheus.Collector.
+func (w *dynamoWatcher) Collect(ch chan<- prometheus.Metric) {
+	w.accountMaxCapacity.Collect(ch)
+	w.tableCapacity.Collect(ch)
+}
```
Comment (tomwilkie) [inline chunk/dynamo_client.go]: You could use a global gauge for this; this style is very verbose.

Comment (jml) [inline chunk/dynamo_client.go]: Can you & @juliusv please sort this out between yourselves? I don't have a strong preference either way.

Comment (juliusv) [inline chunk/dynamo_client.go]: I favor encapsulated metrics that don't poop all over your global package namespace (was annoying in the past when reusing a package, especially when they also register themselves globally, but then aren't even set properly or aren't always relevant for the user of a package), but I don't care enough to fight about it here, and reuse of Cortex packages is less likely. Do it any way you guys prefer, you'll have to live with it longer :)

Comment (tomwilkie) [inline chunk/dynamo_client.go]: I strongly favour global metrics, mainly as its less code, and its harder to forget to register them! (https://github.com/weaveworks/cortex/commit/140c8225602ac21c3f5c106e8ff4b57c4f2a8f4d)

I seem to remember something about custom registries?  Could each component have its own registry, and can you register and registry to another registry?

Comment (tomwilkie) [inline chunk/dynamo_client.go]: PS maybe lets not let this discussion hold up this PR

**Comment by jml** [comment]
Parametrized polling interval (and fixed a lint warning while I was there).

Leaving metrics as-is.

**Pull request merged**